runtime/src/tokio/telemetry.rs

```diff
@@ -26,7 +27,21 @@ pub fn init(context: Context, level: Level, metrics: Option<SocketAddr>, traces:
     if let Some(cfg) = metrics {
         context
             .with_label("metrics")
-            .spawn(move |context| async move { metrics::server::serve(context, cfg).await });
+            .spawn(move |context| async move {
+                let listener = context
+                    .bind(cfg)
+                    .await
+                    .expect("Could not bind to metrics address");
+                let app = Router::new()
+                    .route(
+                        "/metrics",
+                        get(|extension: Extension<Context>| async move { extension.0.encode() }),
+                    )
+                    .layer(Extension(context));
+                serve(listener, app.into_make_service())
```

Comment:
`serve` uses its own reference to the `tokio::spawn` rather than accepting a spawner.